gc/base/ParallelDispatcher.cpp

```diff
@@ -532,12 +524,15 @@ MM_ParallelDispatcher::prepareThreadsForTask(MM_EnvironmentBase *env, MM_Task *t
 {
 	omrthread_monitor_enter(_workerThreadMutex);
 	
+	uintptr_t activeThreads = recomputeActiveThreadCountForTask(env, task, threadCount);
+	task->mainSetup(env);
+
 	/* Set _workerThreadsReservedForGC to true so that shutdown will not 
 	 * attempt to kill the worker threads until after this task is completed
 	 */
 	_workerThreadsReservedForGC = true; 
 
-	Assert_MM_true(_task == NULL);
+	Assert_MM_true(NULL == _task);
 	_task = task;
 
 	task->setSynchronizeMutex(_synchronizeMutex);
```

Comment:
potential simplification (not part of this change): Task's _synchronizeMutex could be set during Task construction, since Dispatcher is already known at that point